src/Microsoft.OData.Core/UriParser/Binders/RangeVariableBinder.cs

```diff
@@ -24,6 +25,16 @@ internal static SingleValueNode BindRangeVariableToken(RangeVariableToken rangeV
         {
             ExceptionUtils.CheckArgumentNotNull(rangeVariableToken, "rangeVariableToken");
 
+            // If $this is the rangeVariableToken Name, we push a rangeVariable with the same TypeReference as the ImplicitRangevariable
+            // to the state.RangeVariables stack.
+            if (rangeVariableToken.Name == ExpressionConstants.This)

```

Comment:
>ExpressionConstants.This [](start = 43, length = 24)

how can we do if we support "non-dollar-sign-prefix" query? 
for example, we support filter same as $filter, how about this same as $this? 

also same question for @mikepizzo

Reply:
I thought "non-dollar-sign-prefix" applies to system query options like $filter, $select etc not in literals like $root, $it and $this

Reply:
no-dollar-prefix does not apply to $this/$it.  It is only for query options ($filter, $orderby, $expand, etc.)

---
In reply to: [577031357](https://github.com/OData/odata.net/pull/1998#discussion_r577031357) [](ancestors = 577031357)

Reply:
exactly.

---
In reply to: [577789986](https://github.com/OData/odata.net/pull/1998#discussion_r577789986) [](ancestors = 577789986)